client/consensus/grandpa/justification.go

```diff
@@ -48,7 +48,7 @@ func NewJustificationFromCommit[
 	H Header[Hash, N]](
```

Comment:
You shouldn't need to have to use a generic parameter for the type `Header`. Header can be defined inside the struct as `Header[Hash, N]`.

Reply:
To clarify, do you mean instead of using generics for  `Header ` I set the type of VoteAncestries to use `Header[Hash, N]`?

Since we do not have a concrete `Header` type in grandpa and we use an interface, this would mean setting the expected type to be an interface. I think this would be okay except that we need to be able to encode/decode the `GrandpaJustification` to the db, which from my experience does not work in this manner. 

However, if there is a way to do this that I am not aware of let me know and I can make this change.

Reply:
Encoding an interface should work.  SCALE uses `reflect.ValueOf` which gives the underlying the type ([playground ex.](https://go.dev/play/p/mklaCRumc_3)).

Reply:
With the work done in https://github.com/ChainSafe/gossamer/pull/3617 I was able to use the custom decoding helper function to make this work